generators/csharp/model/src/snippets/ExampleGenerator.ts

```diff
@@ -0,0 +1,173 @@
+import { csharp } from "@fern-api/csharp-codegen";
+import {
+    ExampleContainer,
+    ExampleNamedType,
+    ExampleObjectType,
+    ExamplePrimitive,
+    ExampleTypeReference,
+    ExampleUndiscriminatedUnionType,
+    ExampleUnionType
+} from "@fern-fern/ir-sdk/api";
+import { ModelGeneratorContext } from "../ModelGeneratorContext";
+import { ObjectGenerator } from "../object/ObjectGenerator";
+
+export class ExampleGenerator {
+    private context: ModelGeneratorContext;
+
+    constructor(context: ModelGeneratorContext) {
+        this.context = context;
+    }
+
+    public getSnippetForTypeReference(exampleTypeReference: ExampleTypeReference): csharp.CodeBlock {
+        const astNode = exampleTypeReference.shape._visit<csharp.AstNode>({
+            primitive: (primitive) => this.getSnippetForPrimitive(primitive),
+            container: (container) => this.getSnippetForContainer(container),
+            unknown: (value) => this.getSnippetForUnknown(value),
+            named: (named) => this.getSnippetForNamed(named),
+            _other: () => {
+                throw new Error("Unknown example type reference: " + exampleTypeReference.shape.type);
+            }
+        });
+        return csharp.codeblock((writer) => writer.writeNode(astNode));
+    }
+
+    private getSnippetForUnknown(unknownExample: unknown): csharp.AstNode {
+        switch (typeof unknownExample) {
+            case "boolean":
+                return csharp.InstantiatedPrimitive.boolean(unknownExample);
+            case "string":
+                return csharp.InstantiatedPrimitive.string(unknownExample);
+            case "number":
+                return csharp.InstantiatedPrimitive.double(unknownExample);
+            case "object":
+                if (Array.isArray(unknownExample)) {
+                    const values = unknownExample.map((value) => this.getSnippetForUnknown(value));
+                    return csharp.list({ entries: values, itemType: csharp.Type.optional(csharp.Type.object()) });
+                } else if (unknownExample != null && unknownExample instanceof Object) {
+                    const keys = Object.keys(unknownExample).sort();
+                    const entries = keys.map((key) => ({
+                        key: csharp.InstantiatedPrimitive.string(key),
+                        value: this.getSnippetForUnknown((unknownExample as Record<string, unknown>)[key])
+                    }));
+                    return csharp.dictionary({
+                        keyType: csharp.Type.object(),
+                        valueType: csharp.Type.optional(csharp.Type.object()),
+                        entries
+                    });
+                }
+                break;
+        }
+        return csharp.InstantiatedPrimitive.null();
```

Comment:
cc @amckinney ideally the IR supports `unknown` examples in a first class way instead of the generator author having to parse an `unknown` themselves